packages/story-editor/src/components/form/dateTime/timePicker.js

```diff
@@ -171,20 +177,20 @@ function TimePicker({ onChange, is12Hour, localeData, setLocaleData }) {
               step={1}
               min={getMinHours()}
               max={getMaxHours()}
-              value={parseInt(localeData.hours)}
+              value={localeData.hours}
               onChange={onChangeEvent('hours')}
+              updateOnChange
               onBlur={updateHours}
-              padZero
```

Comment:
Curious, why remove `padZero` again here and below?

Reply:
I intentionally removed `padZero`. I was experimenting with and without padding. It seems, we may skip padding in hours but required in minutes.

I was experimenting to mach padding with the textbox as shown below:
<img width="376" alt="image" src="https://github.com/GoogleForCreators/web-stories-wp/assets/75766877/a98659ad-235f-458e-91b5-c2fa97014018">

What do you say? Should I enable back padding for both OR just for minutes?

Reply:
> I was experimenting to mach padding with the textbox as shown below:

That is a good idea! Let's try to match padding with the time format in the textbox.

The time format is coming from Settings -> General -> Time Format.

Possible formats, as per PHP documentation:

g | Hour, 12-hour, without leading zeros | 1–12
-- | -- | --
h | Hour, 12-hour, with leading zeros | 01–12
G | Hour, 24-hour, without leading zeros | 0-23
H | Hour, 24-hour, with leading zeros | 00-23
i | Minutes, with leading zeros | 00-59

So the minutes always have leading zeros, so we can force padding there.

For the hours, `h` and `H` force leading zeros, `g` and `G` don't.

So to accommodate for that, we could introduce a new utility function similar to [`is12Hour`](https://github.com/GoogleForCreators/web-stories-wp/blob/a682e9be42e621ebfbefeedb04c40ba44ac03dc7/packages/date/src/is12Hour.ts#L22) to detect whether the hour has leading zeros or not. Maybe `hasLeadingZeros.ts`?

Then we could basically do `padZero={hasLeadingZeros()}`

What do you think? :-)

Reply:
Awesome! That's a great idea.

> So to accommodate for that, we could introduce a new utility function similar to [is12Hour](https://github.com/GoogleForCreators/web-stories-wp/blob/a682e9be42e621ebfbefeedb04c40ba44ac03dc7/packages/date/src/is12Hour.ts#L22) to detect whether the hour has leading zeros or not. Maybe `hasLeadingZeros.ts`?

Yes, `hasLeadingZeros.ts` utility sounds good 👍🏻 and shall implement accordingly.

Reply:
@swissspidy Instead of introducing a new utility function, may we use `is12Hour` like this for hour?

```diff
- padZero={hasLeadingZeros}
+ padZero={!is12Hour}
```

I think it seems much easy to understand. What do you suggest?

Reply:
Unfortunately not, because you can use 12 hour format both with and without leading zeros:

g | Hour, 12-hour, without leading zeros | 1–12
-- | -- | --
h | Hour, 12-hour, with leading zeros | 01–12
G | Hour, 24-hour, without leading zeros | 0-23
H | Hour, 24-hour, with leading zeros | 00-23
i | Minutes, with leading zeros | 00-59

So is12Hour alone does not tell us whether padding is needed or not

Reply:
> Unfortunately not, because you can use 12 hour format both with and without leading zeros:
> 
> g	Hour, 12-hour, without leading zeros	1–12
> h	Hour, 12-hour, with leading zeros	01–12
> G	Hour, 24-hour, without leading zeros	0-23
> H	Hour, 24-hour, with leading zeros	00-23
> i	Minutes, with leading zeros	00-59
> So is12Hour alone does not tell us whether padding is needed or not

Ohh! I missed that. I am updating accordingly.